packages/liveblocks-node/src/client.ts

```diff
@@ -929,6 +1016,256 @@ export class Liveblocks {
+      }
+    );
+    if (!res.ok) {
+      const text = await res.text();
+      throw new LiveblocksError(res.status, text);
+    }
+    return this.convertToCommentData((await res.json()) as CommentDataOriginal);
+  }
+
+  /**
+   * Edits a comment.
+   * @param params.roomId The room ID to edit the comment in.
+   * @param params.threadId The thread ID to edit the comment in.
+   * @param params.commentId The comment ID to edit.
+   * @param params.data.body The body of the comment.
+   * @param params.data.editedAt (optional) The date the comment was edited.
+   * @returns The edited comment.
+   */
+  public async editComment(params: {
+    roomId: string;
+    threadId: string;
+    commentId: string;
+    data: {
+      body: CommentBody;
+      editedAt?: Date;
+    };
+  }): Promise<CommentData> {
+    const { roomId, threadId, commentId, data } = params;
+
+    const res = await this.post(
+      url`/v2/rooms/${roomId}/threads/${threadId}/comments/${commentId}`,
+      {
+        ...data,
+        editedAt: data.editedAt?.toISOString(),
+      }
+    );
+    if (!res.ok) {
+      const text = await res.text();
+      throw new LiveblocksError(res.status, text);
+    }
+
+    return this.convertToCommentData((await res.json()) as CommentDataOriginal);
+  }
+
+  /**
+   * Deletes a comment.
+   * @param params.roomId The room ID to delete the comment in.
+   * @param params.threadId The thread ID to delete the comment in.
+   * @param params.commentId The comment ID to delete.
+   */
+  public async deleteComment(params: {
+    roomId: string;
+    threadId: string;
+    commentId: string;
+  }): Promise<void> {
+    const { roomId, threadId, commentId } = params;
+
+    const res = await this.delete(
+      url`/v2/rooms/${roomId}/threads/${threadId}/comments/${commentId}`
+    );
+    if (!res.ok) {
+      const text = await res.text();
+      throw new LiveblocksError(res.status, text);
+    }
+  }
+
+  /**
+   * Creates a new thread. The thread will be created with the specified comment as its first comment.
+   * If the thread already exists, a `LiveblocksError` will be thrown with status code 409.
+   * @param params.roomId The room ID to create the thread in.
+   * @param params.thread.metadata (optional) The metadata for the thread. Supports upto a maximum of 10 entries. Value must be a string, boolean or number
+   * @param params.thread.comment.userId The user ID of the user who created the comment.
+   * @param params.thread.comment.createdAt (optional) The date the comment was created.
+   * @param params.thread.comment.body The body of the comment.
+   * @returns The created thread. The thread will be created with the specified comment as its first comment.
+   */
+  public async createThread<
+    TThreadMetadata extends BaseMetadata = never,
+  >(params: {
+    roomId: string;
+    data: {
+      metadata?: [TThreadMetadata] extends [never]
+        ? Record<string, never>
```

Comment:
@CTNicholas @nvie @FlowFlorent Any suggestion on the name of this variable. The information in this variable is generally used in the request body. I've currently named it `data` but we use the name `params` to describe request body in `createRoom` method, so I am worried about inconsistency here. Would appreciate your thoughts:

For context, this is the implementation of `createRoom` method:

```ts
public async createRoom(
  roomId: string,
  params: {
    defaultAccesses: RoomPermission;
    groupsAccesses?: RoomAccesses;
    usersAccesses?: RoomAccesses;
    metadata?: RoomMetadata;
  }
): Promise<RoomInfo> {
  const { defaultAccesses, groupsAccesses, usersAccesses, metadata } = params;

  const res = await this.post(url`/v2/rooms`, {
    id: roomId,
    defaultAccesses,
    groupsAccesses,
    usersAccesses,
    metadata,
  });

  ..
}
```

Reply:
I would still go for `data` on other endpoints, rational being that we are nesting the data as part of a `params` object for the comment methods.

`params` = everything to achieve the call
`data`= data specific to the body